### ISSUE TITLE
[dotnet-sdk-7.0.100-rc.2.22452.3] Microsoft.NETCore.App runtime package has mismatched version

### ISSUE THREAD

**Issue by Junjun-zhao** [dotnet/sdk#27675]
**Application Name**:  All WPF and WinForm apps (Total 172 apps)
**OS**: Windows 10 RS5
**CPU**: X64
**.NET Build Number:** dotnet-sdk-7.0.100-rc.2.22452.3
DevDiv Bug checking at: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1611604
 
**Verify Scenarios**:
1). Windows10 RS5 x64+dotnet-sdk-7.0.100-rc.2.22452.3: Fail
2). Windows10 RS5 x64+dotnet-sdk-7.0.100-rc.1.22429.3: Pass

**Repro Steps:**
**The machine only has dotnet-sdk-7.0.100-rc.2.22452.3 installed and set DOTNET_ROLL_FORWARD_ON_NO_CANDIDATE_FX environment variable as 2.**
1. Run this command to create a WPF project with default project template: dotnet new wpf
2. Run this command to run the project: dotnet run

**Expected Result**: 
WPF project will run successfully.

**Actual Result:**
A prompt window abut "You must install or update .NET to run this application" appear and project cannot run.
Image

**Findings:**
In file: "C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\7.0.0-rc.2.22426.1\Microsoft.WindowsDesktop.App.runtimeconfig.json"
```
{
  "runtimeOptions": {
    "tfm": "net7.0",
    "rollForward": "LatestPatch",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "7.0.0-rc.2.22425.17"
    },
    "configProperties": {
      "System.Reflection.Metadata.MetadataUpdater.IsSupported": false
    }
  }
}
```
Actual installed version is:
```Microsoft.NETCore.App 7.0.0-rc.1.22422.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]```

After pointing to the installed version (7.0.0-rc.1.22422.12), the apps can be launch as normal.

**.NET SDK build info**:
```
.NET SDK:
 Version:   7.0.100-rc.2.22452.3
 Commit:    feb4a05460

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19044
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.100-rc.2.22452.3\

Host:
  Version:      7.0.0-rc.1.22422.12
  Architecture: x64
  Commit:       ef077d0b58

.NET SDKs installed:
  7.0.100-rc.2.22452.3 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.0-rc.2.22426.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.0-rc.1.22422.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 7.0.0-rc.2.22426.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

@dotnet-actwx-bot @dotnet/compat

**Comment by PriyaPurkayastha** [comment]
@mmitche this issue was found as part of our nightly runs using Main. Is this expected? Should we hold off testing from this branch?

**Comment by Junjun-zhao** [comment]
Verified and fixed on dotnet-sdk-7.0.100-rc.2.22459.2， closing it.
